gobblin-core/src/main/java/org/apache/gobblin/converter/grok/GrokToJsonConverter.java

```diff
@@ -0,0 +1,223 @@
+ * limitations under the License.
+ */
+package org.apache.gobblin.converter.grok;
+
+import java.io.FileInputStream;
+import java.io.FileNotFoundException;
+import java.io.InputStreamReader;
+import java.io.UnsupportedEncodingException;
+
+import org.apache.commons.lang3.StringUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.common.annotations.VisibleForTesting;
+import com.google.common.base.Preconditions;
+import com.google.gson.JsonArray;
+import com.google.gson.JsonElement;
+import com.google.gson.JsonNull;
+import com.google.gson.JsonObject;
+import com.google.gson.JsonParser;
+
+import io.thekraken.grok.api.Grok;
+import io.thekraken.grok.api.Match;
+import io.thekraken.grok.api.exception.GrokException;
+
+import org.apache.gobblin.configuration.WorkUnitState;
+import org.apache.gobblin.converter.Converter;
+import org.apache.gobblin.converter.DataConversionException;
+import org.apache.gobblin.converter.SchemaConversionException;
+import org.apache.gobblin.converter.SingleRecordIterable;
+
+
+/**
+ * GrokToJsonConverter accepts already deserialized text row, String, where you can use.
+ *
+ * Converts Text to JSON based on Grok pattern. Schema is represented by the form of JsonArray same interface being used by CsvToJonConverter.
+ * Each text record is represented by a String.
+ * The converter only supports Grok patterns where every group is named because it uses the group names as column names.
+ *
+ * The following config properties can be set:
+ * The grok pattern to use for the conversion:
+ * converter.grokToJsonConverter.grokPattern ="^%{IPORHOST:clientip} (?:-|%{USER:ident}) (?:-|%{USER:auth}) \[%{HTTPDATE:timestamp}\] \"(?:%{WORD:verb} %{NOTSPACE:request}(?: HTTP/%{NUMBER:httpversion})?|-)\" %{NUMBER:response} (?:-|%{NUMBER:bytes})"
+ *
+ * Path to the file which contains the base grok patterns which can be used in the converter's GROK pattern (if not set it will use the default ones):
+ * converter.grokToJsonConverter.baseGrokPatternsFile=
+ *
+ * Treat empty string as json null value. The string is empty string if it contains only whitespace characters or empty ("") If it is true the empty string will be converted to json null:
+ * converter.grokToJsonConverter.emptyAsNull=true
+ *
+ * Specify the null string. The specified string will be json null in the converted json:
+ * converter.grokToJsonConverter.nullString="null"
+ *
+ * Example of schema:
+ * [
+  {
+    "columnName": "Day",
+    "comment": "",
+    "isNullable": "true",
+    "dataType": {
+      "type": "string"
+    }
+  },
+  {
+    "columnName": "Pageviews",
+    "comment": "",
+    "isNullable": "true",
+    "dataType": {
+      "type": "long"
+    }
+  }
+]
+ */
+public class GrokToJsonConverter extends Converter<String, JsonArray, String, JsonObject> {
+
+  private static final Logger LOG = LoggerFactory.getLogger(GrokToJsonConverter.class);
+  private static final JsonParser JSON_PARSER = new JsonParser();
```

Comment:
Sorry to mislead you earlier. I found a json parser is schema specific and it can't be shared to parse a second schema if the two have some record field type definitions in common.

Reply:
Sorry, I didn't get it fully.

Reply:
What I mean is "one schema parser per schema" instead of "a static schema parser for all". Because I found static parser caused some error in my work.

Reply:
Sorry. I forgot that it's a json parser not a schema parser. There is no issue.